aten/src/ATen/core/ivalue_inl.h

```diff
@@ -426,6 +435,18 @@ std::vector<Elem> generic_to(
   }
   return result;
 }
+template <typename T>
+T generic_to(
+    IValue ivalue,
+    _fake_type<T>) {
+    auto obj = ivalue.toObject();
+    auto capsule = obj->getAttr("capsule");
+    auto &capsulePtr = capsule.toCapsule()->ptr;
+    if (capsulePtr == nullptr) {
+      capsulePtr = malloc(sizeof(typename std::remove_pointer<T>::type));
+    }
+    return reinterpret_cast<T>(capsule.toCapsule()->ptr);
```

Comment:
Another advantage of using `Blob` instead of `Capsule` is that this cast can be checked properly, since we hold the original type.